runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/UnboundedReadFromBoundedSource.java

```diff
@@ -127,9 +131,11 @@ public void validate() {
         long desiredBundleSize = boundedSource.getEstimatedSizeBytes(options) / desiredNumSplits;
         if (desiredBundleSize <= 0) {
           LOG.warn(
-              "BoundedSource {} cannot estimate its size, skips the initial splits.",
```

Comment:
Your comment makes sense. To make matters more complicated our API instructs source authors to return 0L if size of source cannot be determined. 

So I updated code to pick a default size of 64MB in case 0 is returned as the estimated source. Otherwise we'll try to maximize splitting by setting desiredBindleSize to 1 when a valid value cannot be determined.